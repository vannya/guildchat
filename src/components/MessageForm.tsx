import React, { ChangeEvent, SyntheticEvent } from 'react';
export interface IMessageFormProps {
	id: string;
	buttonText?: string;
	placeholderText?: string;
	value: string;
	disabled?: boolean;
	onClick: (e: SyntheticEvent) => void;
	onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}
export const MessageForm = ({
	id,
	buttonText = 'Send',
	placeholderText = 'Enter Text',
	value,
	disabled = false,
	onClick,
	onChange
}: IMessageFormProps): JSX.Element => {
	return (
		<form className='message-form'>
			<input
				id={id}
				type='text'
				onChange={onChange}
				value={value}
				maxLength={500}
				placeholder={placeholderText}
			/>
			<button
				className='message-form--button'
				disabled={disabled}
				onClick={onClick}
			>
				{buttonText}
			</button>
		</form>
	);
};
