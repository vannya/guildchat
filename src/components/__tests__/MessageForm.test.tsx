import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { MessageForm, IMessageFormProps } from '../MessageForm';

describe('MessageForm', () => {
	const getProps = (): IMessageFormProps => {
		return {
			id: 'message-chat',
			value: '',
			onClick: jest.fn(e => e.preventDefault()),
			onChange: jest.fn()
		};
	};

	it('should match snapshot', () => {
		const props = getProps();
		const { container } = render(<MessageForm {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should calls onChange', () => {
		const props = getProps();
		const { container } = render(<MessageForm {...props} />);
		const inputField = container.querySelectorAll(`#${props.id}`);
		expect(inputField).toHaveLength(1);
		userEvent.type(inputField[0], 'Test Message');
		expect(props.onChange).toHaveBeenCalledTimes(12); // once per char
		cleanup();
	});

	it('should calls onClick', () => {
		const props = getProps();
		const { container } = render(<MessageForm {...props} />);
		const button = container.querySelectorAll(`.message-form--button`);
		expect(button).toHaveLength(1);
		userEvent.click(button[0]);
		expect(props.onClick).toHaveBeenCalledTimes(1);
		cleanup();
	});
});
