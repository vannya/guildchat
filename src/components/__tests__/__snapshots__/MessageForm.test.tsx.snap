// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should match snapshot 1`] = `
<div>
  <form
    class="message-form"
  >
    <input
      id="message-chat"
      maxlength="500"
      placeholder="Enter Text"
      type="text"
      value=""
    />
    <button
      class="message-form--button"
    >
      Send
    </button>
  </form>
</div>
`;
